### PR TITLE
Get projects from memsource with paginated response, added all pages

### PR DESCRIPTION
**Pull request by adityamulik**
Initially, the response was giving output for only 1 page of all the projects, now have added to get a list of all projects from all the pages of totalPages of the paginated API response.

### CODE REVIEW

**Comment by adityamulik** [comment]
@Spredzy I've updated the pre-translate flag and updated the requests.post() method to push any binary file to memsource (e.g. json file upload to memsource).

File: memsource/__init__.py [hunk 1]
```diff
@@ -247,13 +261,12 @@ def create_job(self, project_id, langs, filename, split_filename=None, **kwargs)
             _filename = filename
 
         headers = {
-            'Content-type': 'application/octet-stream',
```
Comment (rooftopcellist) [inline memsource/__init__.py:250]: @adityamulik found that a binary stream does not work for json data, but not specifying the content type header works as expected.

**Review by rooftopcellist — APPROVED** [review]

File: memsource/__init__.py [hunk 2]
```diff
@@ -247,13 +261,14 @@ def create_job(self, project_id, langs, filename, split_filename=None, **kwargs)
...
             'Content-Disposition': 'filename=' + _filename,
             'Memsource': json.dumps(kwargs),
         }
 
+        headers.update({"Authorization": "ApiToken %s" % self.auth.token})
+
         try:
-            return self.handle_rest_call(url, "POST", data=kwargs, headers=headers, files=files).json()
+            return requests.post(url, headers=headers, data=files).json() # Bug fix to push json files
```
Comment (Spredzy) [inline memsource/__init__.py:271]: @adityamulik is there a reason `self.handle_rest_call` couldn't be used here ? Could you expand it (if need be) so it supports what you are trying to do ?

File: memsource/__init__.py [hunk 3]
```diff
@@ -247,13 +261,14 @@ def create_job(self, project_id, langs, filename, split_filename=None, **kwargs)
             _filename = filename
 
         headers = {
-            'Content-type': 'application/octet-stream',
             'Content-Disposition': 'filename=' + _filename,
             'Memsource': json.dumps(kwargs),
         }
 
+        headers.update({"Authorization": "ApiToken %s" % self.auth.token})
```
Comment (Spredzy) [inline memsource/__init__.py:268]: This should be handled by `self.handle_rest_call()` IIRC

File: memsource/__init__.py [hunk 4]
```diff
@@ -229,16 +243,16 @@ def create_project_from_template(self, name, template_id, **kwargs):
...
             raise exc
 
-    def create_job(self, project_id, langs, filename, split_filename=None, **kwargs):
+    def create_job(self, project_id, langs, filename, split_filename=None, preTranslate=True, **kwargs):
         """Create a Memsource job for a given project and lang"""
 
         url = "%s/projects/%s/jobs" % (MEMSOURCE_ENDPOINT_V1_URL, project_id)
 
         kwargs.update({'targetLangs': langs})
+        kwargs.update({'preTranslate': preTranslate})
```
Comment (Spredzy) [inline memsource/__init__.py:252]: nit: You chould used the above update and just expand on the dict.

Currently you have:

```
kwargs.update({'targetLangs': langs})
kwargs.update({'preTranslate': preTranslate})
```

You could go

```
kwargs.update({
    'targetLangs': langs.
    'preTranslate': preTranslate
})
```

**Review by Spredzy — CHANGES_REQUESTED** [review]
Some inner comments before merging

File: memsource/__init__.py (hunk 4 above)
Comment (adityamulik) [reply memsource/__init__.py]: Updated the kwargs as per above.

File: memsource/__init__.py [hunk 5]
```diff
@@ -121,12 +133,24 @@ def delete_import_settings(self, import_setting_id, do_not_fail_on_404=False):
...
+        Each paginated response is binded
+        to a list and returned
+        Return type: list
+        """
 
         url = "%s/projects" % MEMSOURCE_ENDPOINT_V1_URL
 
         try:
             projects = self.handle_rest_call(url, "GET").json()["content"]
+            total_pages = self.handle_rest_call(url, "GET").json()["totalPages"]
```
Comment (rooftopcellist) [inline memsource/__init__.py:147]: Passing thought, if we end up having more instances where we need to get all pages, maybe we could make this a separate function that takes URL path as an argument and returns the content of all pages.

**Review by Spredzy — APPROVED** [review]

**Pull request merged**